### PR TITLE
add scorpius_m32 in keyboard option.

### PR DESCRIPTION
**Pull request by Vencer-Co**
## Types of Changes

- [x] Core
- [x] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Please review the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and verify that you have followed the checklist for the type of PR this is.

**Comment by Vencer-Co** [comment]
I want to add the keyboard in options
截圖 2022-05-03 下午7 09 46
.

**Comment by drashna** [comment]
That is understood.  However, there are a number of changes that need to be made, that are outlined in the PR checklist document. 

Namely, license headers, and such.

**Comment by Vencer-Co** [comment]
how to apply the license

File: keyboards/scorpius_m32/config.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#pragma once
```
Comment (drashna) [inline keyboards/scorpius_m32/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/scorpius_m32/keymap.c [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/scorpius_m32/keymap.c:1]: This file needs to be removed.

File: keyboards/scorpius_m32/keymaps/default/config.h [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#pragma once
```
Comment (drashna) [inline keyboards/scorpius_m32/keymaps/default/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15


ALso, this should probably be moved to the via folder, or set in the keyboard's config.h file

File: keyboards/scorpius_m32/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/scorpius_m32/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/scorpius_m32/readme.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# Scorpius_m32
+
+![Scorpius_m32](https://deskthority.net/wiki/images/e/ed/HHKB_Lite_2.jpg)
+
+This is for the [Scorpius_m32](https://deskthority.net/wiki/HHKB_Lite)
+by PFU, identified by model numbers PD-KB200W/U or PD-KB200B/U. It
+uses the replacement Teensy 2.0 board from [my other
+project](https://github.com/thirteen37/HHKB-Lite-2-Teensy).
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/scorpius_m32/readme.md:10]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/scorpius_m32/rules.mk [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+UNICODE_ENABLE = yes
```
Comment (drashna) [inline keyboards/scorpius_m32/rules.mk:19]: This shouldn't be enabled at the keyboard level. 
```suggestion
```

File: keyboards/scorpius_m32/scorpius_m32.c [hunk 7]
```diff
@@ -0,0 +1 @@
+#include "scorpius_m32.h"
```
Comment (drashna) [inline keyboards/scorpius_m32/scorpius_m32.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/scorpius_m32/scorpius_m32.h [hunk 8]
```diff
@@ -0,0 +1,22 @@
+#pragma once
```
Comment (drashna) [inline keyboards/scorpius_m32/scorpius_m32.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/scorpius_m32/keymaps/default/config.h (hunk 3 above)
Comment (drashna) [reply keyboards/scorpius_m32/keymaps/default/config.h:4]: > Also, this should probably be moved to the via folder, or set in the keyboard's config.h file

File: keyboards/scorpius_m32/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+// Copyright 2022 Vencer-Co (@Vencer-Co)
+// SPDX-License-Identifier: GPL-2.0-or-later
```
Comment (drashna) [inline keyboards/scorpius_m32/keymaps/default/keymap.c:4]:
```suggestion

// Copyright 2022 Vencer-Co (@Vencer-Co)
// SPDX-License-Identifier: GPL-2.0-or-later

#include QMK_KEYBOARD_H

```

**Comment by Vencer-Co** [comment]
Next, What i do something for merging the project?

File: keyboards/scorpius_m32/keymaps/default/config.h (hunk 3 above)
Comment (Vencer-Co) [reply keyboards/scorpius_m32/keymaps/default/config.h]: I have finished the fix.

**Review by fauxpark — APPROVED** [review]

File: keyboards/scorpius_m32/config.h [hunk 10]
```diff
@@ -0,0 +1,32 @@
+
+// Copyright 2022 Vencer-Co (@Vencer-Co)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x195D
```
Comment (drashna) [inline keyboards/scorpius_m32/config.h:9]: This vender ID is in use, and needs to be changed. 
https://www.the-sz.com/products/usbid/index.php?v=0x195D&p=&n=

Generally, anything above 0x4000 is good, but the higher, the better.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/scorpius_m32/config.h (hunk 10 above)
Comment (Vencer-Co) [reply keyboards/scorpius_m32/config.h:9]: The VENDOR_ID 0x195D is belongs to my company.

File: keyboards/scorpius_m32/readme.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
+# Scorpius_m32
+
+![Scorpius_m32](https://vencer-co.github.io/scorpius_m32/images/m32.png)
```
Comment (drashna) [inline keyboards/scorpius_m32/readme.md:3]: This needs to be re-hosted on an image hosting site, such as imgur.

**Pull request closed without merging**